fix: resolve timestamp precision bug in team activity log notifications

nehaaprasad opened the pull request:
### Problem
- Items created microseconds after the bookmark timestamp incorrectly appeared as unread due to inconsistent timestamp precision - the code was truncating `created_at` to milliseconds but not the bookmark timestamp.

#35246

### Changes
- Fixed timestamp comparison in both activity log serializers by removing the truncation logic:
- `posthog/api/my_notifications.py`: Updated `MyNotificationsSerializer.get_unread()`
- `posthog/api/activity_log.py`: Updated `ActivityLogSerializer.get_unread()`

- Before: `return bookmark_date < obj.created_at.replace(microsecond=obj.created_at.microsecond // 1000 * 1000)`

- After: `return bookmark_date < obj.created_at`

### How did you test this code?
- Verified the fix handles microsecond precision correctly
- Confirmed both serializers are updated consistently
- Checked no other timestamp truncation issues exist

nehaaprasad committed:
update

nehaaprasad committed:
- Fixed MyNotificationsSerializer.get_unread() method and ActivityLogSerializer.get_unread() method

nehaaprasad committed:
Merge branch 'master' into fix/team-activity-timestamp-precision

pauldambra commented on frontend/src/scenes/session-recordings/playlist/SessionRecordingsPlaylist.tsx:
can you remove this unrelated change :)

pauldambra commented on posthog/api/my_notifications.py:
```diff
@@ -37,9 +37,10 @@ def get_unread(self, obj: ActivityLog) -> bool:
+           
+            # The frontend sends microseconds via toISOString() and we should respect that precision
             bookmark_date = user_bookmark.last_viewed_activity_date
-            return bookmark_date < obj.created_at.replace(microsecond=obj.created_at.microsecond // 1000 * 1000)
```
we should add a test here that the case in the issue is covered by this fix

if it's not already covered then we should be covering

* something before the timestamp is considered read
* something at the timestamp exactly is considered read
* something after the timestamp is considered unread

i also wonder if we _should_ be truncating the timestamp
if an item was created at 223 microseconds after the bookmark date is it meaningful to say that it is unread?
activity log is not receiving writes at the kind of scale where that accuracy is helpful

maybe the fix is to truncate to the nearest second or half second on both sides of the comparison and have tests protecting that behaviour so the future traveller can see what is happening

naaa760 commented:
I will resolve the issue in some hours!
@pauldambra

nehaaprasad committed:
fix: DRY unread logic and add edge case tests for activity log notifications

- Move unread logic to shared is_activity_unread() utility to remove duplication.
- Update both MyNotificationsSerializer and ActivityLogSerializer to use shared function.
- Add unit tests for before/at/after/None bookmark edge cases.
- Remove unrelated frontend changes.

naaa760 commented:
@pauldambra 
the issues has been resolved sorry for being delay.

nehaaprasad committed:
Merge branch 'master' into fix/team-activity-timestamp-precision
